src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,196 @@
+    private static final String UNKNOWN_ERROR      = "Unknown Error!";
+    private static final String NOT_FOUND          = "%s not found";
+    private static final String NO_COLONY          = "You haven't got a colony!";
+
+
+    public static int colonyId;
+    public static int citizenId;
+
+    public static int getColonyId(@NotNull final ICommandSender sender, @NotNull final String... args)
+    {
+        final UUID mayorID = sender.getCommandSenderEntity().getUniqueID();
+        if (args.length == 2 || args.length == 4)
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                colonyId = -1;
+            }
+        }
+        else if (args.length == 1 || args.length == 3)
+        {
+            try
+            {
+                colonyId = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID).getID();
+            }
+            catch (NullPointerException e)
+            {
+                colonyId = -5;
+            }
+        }
+        else if (args.length >= 5)
+        {
+            colonyId = -2;
+        }
+        else
+        {
+            colonyId = -3;
+        }
+        if (colonyId >= 0)
+        {
+            if (ColonyManager.getColony(colonyId) == null)
+            {
+                colonyId = -4;
+            }
+        }
+
+
+        return colonyId;
+    }
+
+    public static int getCitizenId(@NotNull final int colonyId, @NotNull final String... args)
+    {
+        String citizenName;
+        citizenId = -4;
+        if (args.length == 2)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[1]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = -1;
+            }
+        }
+        else if (args.length == 1)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = -1;
+            }
+        }
+        else if (args.length == 3 && colonyId >= 0)
+        {
+            citizenName = args[0] + " " + args[1] + " " + args[2];
+            for (int i = 1
+                   ; i <= ColonyManager.getColony(colonyId).getCitizens().size(); i++)
+            {
+                try
+                {
+                    if (ColonyManager.getColony(colonyId).getCitizen(i).getName().equals(citizenName))
+                    {
+                        citizenId = i;
+                    }
+                }
+                catch (NullPointerException e)
+                {
+                    continue;
+                }
+            }
+
+        }
+        else if (args.length == 4 && colonyId >= 0)
+        {
+            citizenName = args[1] + " " + args[2] + " " + args[3];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)